### ISSUE TITLE
[Docs] Providers: Clarify Event Persistence Behavior

### ISSUE THREAD

**Issue by leecalcote** [meshery/meshery#13058]
### Current State
Explain the current behavior of the Meshery Server's behavior with respect to local and remote providers and event persistence. 

### Desired State

Update the following pages:
1. https://docs.meshery.io/project/contributing/contributing-server-events
2. https://docs.meshery.io/extensibility/providers

Answer the following question:
1. The provider.PersistEvent(event) does not send events to Remove Providers. Is more control is needed in this regard?

---

### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 📚 [Instructions for contributing to documentation](https://docs.meshery.io/project/contributing/contributing-docs)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/contributing/build-and-release)
- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

**Comment by souvikinator** [comment]
**TLDR;**

No additional control is needed at the persistence level. The current implementation provides a clean separation of concerns:

1. `provider.PersistEvent(event)`: 
        a. For event persistance to local database (Meshery server)
        b. Used by both Local and Remote providers

2. `provider.PublishEventToProvider(token,event)`:
        a. For sending event to a remote provider if the provider supports (ex: Meshery Cloud)
        b. For local provider it's a No-op


**Documentation update needed:**

1. contributing-server-events doc:

- Add dedicated section for event persistence
- Example code for each event persistence methods with explanation

2. providers doc:

- Add event handling capabilities in the _Types of Providers_ section
- Add note about how providers handle event persistence differently (local vs remote)

**Details**

`DefaultLocalProvider` and `RemoteProvider` implement the `Provider` interface and both implement the `PersistEvent(event)` and `PublishEventToProvider(token, event)`

1. `DefaultLocalProvider` ("None" provider) only handles the `PersistEvent(event)`  and `PublishEventToProvider(token, event)` is a No-op

2. `RemoteProvider` handles both `PublishEventToProvider(token, event)` and `PersistEvent(event)`

so no additional control is needed at the persistence level.

**Comment by leecalcote** [comment]
@Deeptanshu-sankhwar
